src/test/java/com/slack/kaldb/elasticsearchApi/ElasticsearchApiServiceTest.java

```diff
@@ -0,0 +1,258 @@
+import com.slack.kaldb.server.KaldbLocalSearcher;
+import com.slack.kaldb.testlib.KaldbConfigUtil;
+import com.slack.kaldb.testlib.MessageUtil;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Optional;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.CreateBucketRequest;
+
+@SuppressWarnings("UnstableApiUsage")
+public class ElasticsearchApiServiceTest {
+  @ClassRule public static final S3MockRule S3_MOCK_RULE = S3MockRule.builder().silent().build();
+  @Rule public final TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  private S3Client s3Client;
+  private ChunkManager<LogMessage> chunkManager;
+  private ElasticsearchApiService elasticsearchApiService;
+
+  private static final String S3_TEST_BUCKET = "test-kaldb-logs";
+  private static final String CHUNK_DATA_PREFIX = "testData";
+
+  @Before
+  public void setUp() throws IOException {
+    KaldbConfigUtil.initEmptyConfig();
+
+    // create an S3 client and a bucket for test
+    s3Client = S3_MOCK_RULE.createS3ClientV2();
+    s3Client.createBucket(CreateBucketRequest.builder().bucket(S3_TEST_BUCKET).build());
+
+    S3BlobFs s3BlobFs = new S3BlobFs();
+    s3BlobFs.init(s3Client);
+
+    chunkManager =
+        new ChunkManager<>(
+            CHUNK_DATA_PREFIX,
+            temporaryFolder.newFolder().getAbsolutePath(),
+            new ChunkRollOverStrategyImpl(10 * 1024 * 1024 * 1024L, 1000000L),
+            new SimpleMeterRegistry(),
+            s3BlobFs,
+            S3_TEST_BUCKET,
+            MoreExecutors.newDirectExecutorService(),
+            3000);
+
+    KaldbLocalSearcher<LogMessage> searcher = new KaldbLocalSearcher<>(chunkManager);
+    elasticsearchApiService = new ElasticsearchApiService(searcher);
+  }
+
+  @After
+  public void tearDown() {
+    chunkManager.close();
+    s3Client.close();
+  }
+
+  // todo - test mapping
+
+  @Test
+  public void testResultsAreReturnedForValidQuery() throws IOException {
+    List<LogMessage> messages = MessageUtil.makeMessagesWithTimeDifference(1, 100);
+    addMessagesToChunkManager(messages);
+
+    String postBody =
+        Resources.toString(
+            Resources.getResource("elasticsearchApi/multisearch_query_500results.ndjson"),
+            Charset.defaultCharset());
+    HttpResponse response = elasticsearchApiService.multiSearch(postBody);
+
+    // handle response
+    AggregatedHttpResponse aggregatedRes = response.aggregate().join();
+    String body = aggregatedRes.content(StandardCharsets.UTF_8);
+    JsonNode jsonNode = new ObjectMapper().readTree(body);
+
+    assertEquals(200, aggregatedRes.status().code());
```

Comment:
For unit tests we use assertj assertions through out the code. Can we please use those instead of junit assertions to keep the code consistent?

Reply:
Ah, I totally didn't catch that - fixed. 👍